iota/api.py

```diff
@@ -22,948 +23,948 @@
-      perform for the network (as it will confirm more transactions
-      that way).
-
-    References:
-      - https://iota.readme.io/docs/gettransactionstoapprove
-    """
-    return core.GetTransactionsToApproveCommand(self.adapter)(depth=depth)
-
-  def get_trytes(self, hashes):
-    # type: (Iterable[TransactionHash]) -> dict
-    """
-    Returns the raw transaction data (trytes) of one or more
-    transactions.
-
-    References:
-      - https://iota.readme.io/docs/gettrytes
+class ApiMeta(type):
     """
-    return core.GetTrytesCommand(self.adapter)(hashes=hashes)
-
-  def interrupt_attaching_to_tangle(self):
-    # type: () -> dict
+    Manages command registries for IOTA API base classes.
     """
-    Interrupts and completely aborts the :py:meth:`attach_to_tangle`
-    process.
+    def __init__(cls, name, bases=None, attrs=None):
+        super(ApiMeta, cls).__init__(name, bases, attrs)
 
-    References:
-      - https://iota.readme.io/docs/interruptattachingtotangle
-    """
-    return core.InterruptAttachingToTangleCommand(self.adapter)()
+        if not hasattr(cls, 'commands'):
+            cls.commands = {}
 
-  def remove_neighbors(self, uris):
-    # type: (Iterable[Text]) -> dict
-    """
-    Removes one or more neighbors from the node.  Lasts until the node
-    is restarted.
+        # Copy command registry from base class to derived class, but
+        # in the event of a conflict, preserve the derived class'
+        # commands.
+        commands = {}
+        for base in bases:
+            if isinstance(base, ApiMeta):
+                commands.update(base.commands)
 
-    :param uris:
-      Use format ``udp://<ip address>:<port>``.
-      Example: `remove_neighbors(['udp://example.com:14265'])`
+        if commands:
+            commands.update(cls.commands)
+            cls.commands = commands
 
-    References:
-      - https://iota.readme.io/docs/removeneighors
-    """
-    return core.RemoveNeighborsCommand(self.adapter)(uris=uris)
 
-  def store_transactions(self, trytes):
-    # type: (Iterable[TryteString]) -> dict
+class StrictIota(with_metaclass(ApiMeta)):
     """
-    Store transactions into local storage.
+    API to send HTTP requests for communicating with an IOTA node.
 
-    The input trytes for this call are provided by
-    :py:meth:`attach_to_tangle`.
+    This implementation only exposes the "core" API methods.  For a more
+    feature-complete implementation, use :py:class:`Iota` instead.
 
     References:
-      - https://iota.readme.io/docs/storetransactions
-    """
-    return core.StoreTransactionsCommand(self.adapter)(trytes=trytes)
+      - https://iota.readme.io/docs/getting-started
+    """
+    commands = discover_commands('iota.commands.core')
+
+    def __init__(self, adapter, testnet=False):
+        # type: (AdapterSpec, bool) -> None
+        """
+        :param adapter:
+          URI string or BaseAdapter instance.
```

Comment:
❔ To be consistent, should we also use 4-space indents inside docstrings?

Reply:
I don't know. Sphinx doesn't respect formatting here anyway, right? Ah, let's make it 4 ok?